### PR TITLE
Fix optional typing introduced in 3.10 and 3.11

### PR DESCRIPTION
**Pull request by tungalbert99**
The existing system triggers in the subcommand which is not correct. You can see without the fix in that the tests will fail.

### CODE REVIEW

File: src/clipstick/_parse.py [hunk 1]
```diff
@@ -1,35 +1,41 @@
...
 def _is_subcommand(attribute: str, field_info: FieldInfo) -> bool:
     """Check if the field annotated as a subcommand."""
+
     if not isinstance(field_info.annotation, UnionType):
         return False
+
     args = get_args(field_info.annotation)
+
+    # Checks to see if the type could be "int | None" for instance
+    if type(None) in args:
```
Comment (sander76) [inline src/clipstick/_parse.py:36]: I'd like to make this more general and have a check like the following:

```python
if not any(issubclass(arg, BaseModel) for arg in args):
        return False
```
this also covers models like this:

```python
class MyModel(BaseModel):
    value: int|str="some value"
```

File: tests/test_user_error_output.py [hunk 2]
```diff
@@ -96,3 +108,19 @@ def test_failing_optional(capture_output, args):
     assert (
         f"Incorrect value for {args[0]} (-10). Input should be greater than 0"
     ) in capture_output.captured_output
+
+
+class OptionalValueOldTyping(BaseModel):
```
Comment (sander76) [inline tests/test_user_error_output.py:113]: The issue you found revolves about cli optional arguments. 

I would say the best place to put any test would be in `test_optionals.py`.

something like this:

```python

class Model(BaseModel):
    value: int | None = None


def test_optional_with_none():
    model = parse(Model, ["--value", "10"])
    assert model == Model(value=10)


class OldModel(BaseModel):
    value: Optional[int] = None


def test_old_model_with_none():
    model = parse(OldModel, ["--value", "10"])
    assert model == OldModel(value=10)
```

**Review by sander76 — COMMENTED** [review]
Thanks for the pr. Good catch!
See my comments.
Currently I have time available, so if you don't, I am happy to take this one over.

**Comment by tungalbert99** [comment]
I can help you with some features in this library! doing this as some after-work fun

File: src/clipstick/_parse.py (hunk 1 above)
Comment (tungalbert99) [reply src/clipstick/_parse.py:36]: good point, done!

File: tests/test_user_error_output.py (hunk 2 above)
Comment (tungalbert99) [reply tests/test_user_error_output.py]: done!

File: tests/test_optionals.py [hunk 3]
```diff
@@ -61,3 +63,19 @@ def test_help_with_shorts(capture_output):
...
     assert err.value.code == 0
     assert "-v --value-1" in capture_output.captured_output
+
+
+class OptionalValueOldTyping(BaseModel):
+    my_first_optional: Optional[int] = None
+
+
+def test_optional_value_old_typing(capture_output):
+    capture_output(OptionalValueOldTyping, [])
```
Comment (sander76) [inline tests/test_optionals.py:73]: Can you change these tests for them to have assertions (and explicit testing) ? Like so:

```python
class Model(BaseModel):
    value: int | None = None

def test_optional_with_none():
    model = parse(Model, ["--value", "10"])
    assert model == Model(value=10)

class OldModel(BaseModel):
    value: Optional[int] = None

def test_old_model_with_none():
    model = parse(OldModel, ["--value", "10"])
    assert model == OldModel(value=10)
```

**Pull request merged**